### PR TITLE
FILTER_TAGS_ISSUE#226

### LINKED ISSUE #226: [FEATURE] add filter section in projects page
### Description

The website can have tags while choosing projects (for example js, ruby, python) Categorizing projects under these tags make it easy for developers to checkout as per their skill set.

### PR DESCRIPTION
**Pull request by matmont**
### Fixes Issue

Closes #226

### Changes proposed

General implementation of the filtering.
I want to propose the editing of the `seed.ts` file. I added a list of technologies (that we could retrieve from some API to have a real full list) and every project will have assigned three random techs.

### CODE REVIEW

**Comment by rohitdasu** [comment]
@matmont this looks awesome! 

just few suggestions to add:
1. can we show a toast message if the typed tag is not present? we are showing red outline in input field, which is good but with a proper message would be better.
2. in mobile view the filter section is touching the boundaries of screen. can you make it flex-col in mobile screens? or anything to make it look better in mobile.

the screenshot is for 320 width mobile:
Screenshot 2023-01-12 at 3 04 25 PM

File: pages/projects/index.tsx [hunk 1]
```diff
@@ -1,20 +1,68 @@
...
+  allProjects.forEach((project) =>
+    project.tags.forEach((tag) => allTags.add(tag))
+  );
+
+  return {
+    props: {
+      availableTags: Array.from(allTags),
+    },
+  };
+}
```
Comment (JakubChorzepa) [inline pages/projects/index.tsx:68]: I think fetching all projects to extract available tags at page request might slow page loading when there would be big amount of projects to fetch.

For now this solution is fine but we could think about adding some back-end logic that will store and track which tags are used in projects and how many times they appear? The tags data could be reached at specific endpoint. I could speed up loading.

What do you think about it? @rohitdasu @matmont

**Comment by rupali-codes** [comment]
Hi, it looks good to me
but, can we just reduce searchbar width a bit(or make it half), because its unnecessarily taking full width. It looks fine for mobile design, but for desktop it needs be reduced.

File: pages/projects/index.tsx (hunk 1 above)
Comment (rohitdasu) [reply pages/projects/index.tsx:68]: i agree

Comment (rohitdasu) [reply pages/projects/index.tsx:68]: in future if we have more tags, it will be slow. need to think of the better solution. as of now it's fine though.

**Comment by rohitdasu** [comment]
@matmont

File: pages/projects/index.tsx (hunk 1 above)
Comment (matmont) [reply pages/projects/index.tsx:68]: yeah I agree, this would also help for the auto-completion when setting tags on projects.

**Review by rohitdasu — APPROVED** [review]
LGTM! thanks @matmont 
It's a good foundation on filters.

**Pull request merged**